sdk/metrics/src/test/java/io/opentelemetry/sdk/metrics/export/IntervalMetricReaderTest.java

```diff
@@ -176,20 +173,12 @@ public CompletableResultCode shutdown() {
      * metrics.
      */
     @Nullable
-    List<List<MetricData>> waitForNumberOfExports(int numberOfExports) {
-      List<List<MetricData>> result;
-      synchronized (monitor) {
-        while (exportedMetrics.size() < numberOfExports) {
-          try {
-            monitor.wait();
-          } catch (InterruptedException e) {
-            // Preserve the interruption status as per guidance.
-            Thread.currentThread().interrupt();
-            return null;
-          }
-        }
-        result = exportedMetrics;
```

Comment:
IIUC, the problem was that exportedMetrics could have more than the expected number of exports returned? Anyways, blockingqueue always better than monitors!

Reply:
Yeah, in some circumstances `export()` was entered a second time before the released `wait()` in the other thread could continue on with its work of adding to the result.  By the time it got around to doing that, there would occasionally be more data than expected.